plugin/registry.go

```diff
@@ -34,10 +34,29 @@ func (r *Registry) Plugins() map[string]Root {
 	return r.plugins
 }
 
-// RegisterPlugin registers the given plugin
-func (r *Registry) RegisterPlugin(name string, root Root) {
+// RegisterPlugin initializes the given plugin and adds it to the registry of
+// initialization was successful.
+func (r *Registry) RegisterPlugin(name string, root Root) error {
+	if err := root.Init(); err != nil {
+		return err
+	}
+
 	r.plugins[name] = root
 	r.pluginRoots = append(r.pluginRoots, root)
+	return nil
+}
+
+// ExternalPluginSpec represents an external plugin's specification.
+type ExternalPluginSpec struct {
+	Name   string
+	Script string
+}
+
+// RegisterExternalPlugin initializes an external plugin from its spec and
+// passes it to RegisterPlugin.
+func (r *Registry) RegisterExternalPlugin(spec ExternalPluginSpec) error {
```

Comment:
Would be useful to check that `spec.Script` exists

Reply:
I did that and added some testing.